### ISSUE TITLE
[bug] `conan lock create` creates `default` profile

### ISSUE THREAD

**Issue by gpxricky** [conan-io/conan#10901]
Hi,

this is not really a blocker, however it seems to be the case that `conan lock create` creates the `default` profile under some circumstances.

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04.6 LTS
  * Compiler+version: <does not matter>
  * Conan version: 1.46.0
  * Python version: 3.8.0

### Steps to reproduce (Include if Applicable)

* Create Directory `test` and change into the directory
* Run `conan new -s test/dev@user/channel`
* Remove `default` profile from Conan cache
* Ensure no default profile is set (e.g. in `conan.conf` or `global.conf`)
* Ensure another profile in the Conan cache is available, e.g. `my_profile`
* Create the lock file with:
   ```
   conan lock create ./conanfile.py \
     --profile:host my_profile \
     --profile:build my_profile \
     --lockfile-out ./output/conan.lock
   ``` 

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
Auto detecting your dev setup to initialize the default profile (/home/user/.conan/profiles/default)
Found gcc 7.5
Found gcc 7.5
gcc>=5, using the major as version
gcc C++ standard library: libstdc++11
Default settings
	os=Linux
	os_build=Linux
	arch=x86_64
	arch_build=x86_64
	compiler=gcc
	compiler.version=7
	compiler.libcxx=libstdc++11
	build_type=Release
*** You can change them in /home/user/.conan/profiles/default ***
*** Or override with -s compiler='other' -s ...s***


Requirements
Packages

Generated lockfile: /home/user/test/output/conan.lock
```

Although the created lock file uses the correct host and build profiles, it is a little bit surprising that the `default` profile will be created.

Best regards,
Markus

**Comment by memsharded** [comment]
Hi @gpxricky 

Just checked, this has already been fixed in 2.0. 
As this isn't really a blocker but mostly UX, and it seems it is not very easy to fix this, I think it is ok to consider it as solved by 2.0 (the currently released alpha seems to work fine already) and close it.

Thanks for reporting!

**Comment by gpxricky** [comment]
Okay, thanks for the info. I will close this bug report then.
